kitsune/users/static/users/js/actions/UserAuthActions.es6.js

```diff
@@ -1,9 +1,132 @@
+/* globals $:false */
+import apiFetch, {apiFetchRaw} from '../../../sumo/js/utils/apiFetch.es6.js';
 import Dispatcher from '../../../sumo/js/Dispatcher.es6.js';
 import {actionTypes} from '../constants/UserAuthConstants.es6.js';
+import UserAuthStore from '../stores/UserAuthStore.es6.js';
 
-export function setAuthState(state) {
+export function checkAuthState() {
+  return apiFetchRaw('/api/1/users/test_auth', {
+    credentials: 'include',
+  })
+  .then(res => {
+    return res.json().then(data => [res.status, data]);
+  })
+  .then(([status, data]) => {
+    if (status === 200) {
+      Dispatcher.dispatch({
+        type: actionTypes.AUTH_LOG_IN_SUCCESS,
+        username: data.username,
+      });
+    } else if (status === 401) {
+      Dispatcher.dispatch({
+        type: actionTypes.AUTH_LOG_OUT,
+      });
+    } else {
+      throw new Error(`Unexpected response from test_auth: ${status}: ${JSON.stringify(data)}`);
+    }
+    return UserAuthStore.getState();
+  });
+}
+
+export function login(username, password, {inactive=false}={}) {
+  const csrfElem = document.querySelector('input[name=csrfmiddlewaretoken]');
+  const csrf = csrfElem ? csrfElem.value : null;
+
+  Dispatcher.dispatch({
+    type: actionTypes.AUTH_LOG_IN_OPTIMISTIC,
+  });
+
+  $.post('/en-US/users/login', {
```

Comment:
I used jquery's ajax function here because it serializes form data nicely. In a perfect world I'd probably have a utility function to do that formatting for me, and  use `fetch` instead.